src/t8_forest/t8_forest_ghost/t8_forest_ghost_implementations/t8_forest_ghost_definition_face.cxx

```diff
@@ -0,0 +1,405 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2025 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_forest_ghost_definition_face.cxx
+ * Implementations for t8_forest_ghost_definition_face.hxx
+ */
+
+#include <t8_forest/t8_forest_ghost/t8_forest_ghost_implementations/t8_forest_ghost_definition_face.hxx>
+#include <t8_forest/t8_forest_ghost/t8_forest_ghost_definition_helpers.hxx>
+#include <t8_schemes/t8_scheme.hxx>
+#include <t8_forest/t8_forest_private.h>
+#include <vector>
+
+/** Struct which holds data for the search of t8_forest_ghost_definition_face */
+struct t8_forest_ghost_definition_face_data: t8_forest_ghost_search_data
+{
+  t8_forest_ghost_definition_face_data ()
+  {
+    sc_array_init (&face_owners, sizeof (int));
+    /* This is a dummy init, since we call sc_array_reset in ghost_search_boundary
+     * and we should not call sc_array_reset on a non-initialized array */
+    sc_array_init (&bounds_per_level, 1);
+  }
+
+  virtual ~t8_forest_ghost_definition_face_data ()
```

Comment:
Do you expect derived classes of t8_forest_ghost_definition_face_data? If not, i would remove the virtual keyword here and add the override keyword